prime-router/src/main/kotlin/Submission.kt

```diff
@@ -0,0 +1,72 @@
+package gov.cdc.prime.router
+
+import java.time.OffsetDateTime
+
+/**
+ * A `Submission` represents one submission of a message from a sender.
+ *
+ * @param taskId of the Submission is `action_id` from table `public.action`
+ * @param createdAt of the Submission is `created_at` from the table `public.action`
+ * @param sendingOrg of the Submission is `sending_org` from the table `public.action`
+ * @param httpStatus of the Submission is `http_status` from the table `public.action`
+ * @param id of the Submission is `action_response.id` and represents a Report ID from the table `public.action`
+ * @param topic of the Submission is `action_response.topic` from the table `public.action`
+ * @param reportItemCount of the Submission is `action_response.reportItemCount` and represents a Report ID from the table `public.action`
+ * @param warningCount of the Submission is `action_response.warningCount` from the table `public.action`
+ * @param errorCount of the Submission is `action_response.errorCount` from the table `public.action`
+ */
+
+/*
+ * TODO: see Github Issues #2314 for expected filename field
+ */
+
+open class SubmissionHistory(
+    val taskId: Long,
+    val createdAt: OffsetDateTime,
+    val sendingOrg: String,
+    val httpStatus: Int,
+    val id: String?,
```

Comment:
Can a submission ever have a null id?

Reply:
Yes. If a submission action fails the Action Response ID will be null.